cmd/controller/main.go

```diff
@@ -23,5 +24,6 @@ import (
 func main() {
 	sharedmain.Main("async-controller",
 		ingress.NewController,
+		service.NewController,
```

Comment:
fwiw Im not sure you'll need a separate reconciler for this. The "kind" you're reconciling is still the top-level ingress, you're just going to spit out a Service as well as delegate-ingress, iiuc

Reply:
I went back and forth on this. I think you're right, it's actually cleaner to keep one reconciler.